### ISSUE TITLE
Unrecognized DAO error related to Crossref Reference Linking Plugin

### ISSUE THREAD

**Issue by AhemNason** [pkp/pkp-lib#6958]
If you have the Crossref Reference Linking plugin enabled on either the most recent 3.2.x release or 3.3.x release, you'll get this error when trying to deposit your DOIs to Crossref. 

> Unrecognized DAO PublishedArticleDAO!

I know there was an Unrecognized DAO error reported by some after the migration to 3.3 but this issue is consistent with users who have the plugin enabled on 3.2.x as well. Disabling the plugin allows submission to occur. I wish I had more for you but this is repeatable and we've had more than one journal in hosting run into it. @israelcefrin may have more information about error logs he's seen on the installation he was troubleshooting.

**Comment by asmecher** [comment]
@AhemNason, I'm pretty sure this means they have a version of the plugin installed that's not compatible with the release of OJS they're using (too old).

**Comment by AhemNason** [comment]
Hey @asmecher, I thought so too, but these installs are showing that this is the latest available version of the plugin. 
![Screen Shot 2021-04-17 at 6 34 37 PM](https://user-images.githubusercontent.com/504276/115127276-ac11be00-9fab-11eb-828c-b7a49e42a0d2.png)

**Comment by asmecher** [comment]
Can someone turn on the stack trace option in config.inc.php and add the resulting stack trace here?

**Comment by NateWr** [comment]
It looks like 1.0.2.1 is only compatible with 3.3.0.4 and 3.3.0.5. https://github.com/pkp/plugin-gallery/blob/main/plugins.xml#L2591-L2598

**Comment by israelcefrin** [comment]
> Can someone turn on the stack trace option in config.inc.php and add the resulting stack trace here?

Hi @asmecher 
Here it goes the stack trace when trying to deposit metadata XML in CrossRef:

```
Unrecognized DAO PublishedArticleDAO!
Stack Trace:
File: /www/plugins/generic/crossrefReferenceLinking/CrossrefReferenceLinkingPlugin.inc.php line 190
Function: DAORegistry::getDAO("PublishedArticleDAO")

File: /www/lib/pkp/classes/plugins/HookRegistry.inc.php line 107
Function: CrossrefReferenceLinkingPlugin->addCrossrefCitationsElements("articlecrossrefxmlfilter::execute", Array(1))

File: /www/lib/pkp/classes/filter/Filter.inc.php line 451
Function: HookRegistry::call("articlecrossrefxmlfilter::execute", Array(1))

File: /www/classes/plugins/PubObjectsExportPlugin.inc.php line 366
Function: Filter->execute(Array(1), True)

File: /www/plugins/importexport/crossref/CrossRefExportPlugin.inc.php line 215
Function: PubObjectsExportPlugin->exportXML(Array(1), "article=>crossref-xml", Object(Journal), True)

File: /www/classes/plugins/PubObjectsExportPlugin.inc.php line 171
Function: CrossRefExportPlugin->executeExportAction(Object(Request), Array(5), "article=>crossref-xml", "exportSubmissions-tab", "articles", True)

File: /www/classes/plugins/DOIPubIdExportPlugin.inc.php line 29
Function: PubObjectsExportPlugin->display(Array(0), Object(Request))

File: /www/lib/pkp/pages/management/PKPToolsHandler.inc.php line 104
Function: DOIPubIdExportPlugin->display(Array(1), Object(Request))

File: /www/lib/pkp/classes/core/PKPRouter.inc.php line 391
Function: PKPToolsHandler->importexport(Array(1), Object(Request))

File: /www/lib/pkp/classes/core/PKPPageRouter.inc.php line 231
Function: PKPRouter->_authorizeInitializeAndCallRequest(Array(2), Object(Request), Array(3), False)

File: /www/lib/pkp/classes/core/Dispatcher.inc.php line 143
Function: PKPPageRouter->route(Object(Request))

File: /www/lib/pkp/classes/core/PKPApplication.inc.php line 281
Function: Dispatcher->dispatch(Object(Request))

File: /www/index.php line 68
Function: PKPApplication->execute()
```

**Comment by bozana** [comment]
Hmmm... it seems to me that the plugin code is still the old one, from the version 1.0.1-1, s. https://github.com/pkp/crossrefReferenceLinking/blob/v1_0_1-1/CrossrefReferenceLinkingPlugin.inc.php#L190 vs. https://github.com/pkp/crossrefReferenceLinking/blob/v1_0_2-1/CrossrefReferenceLinkingPlugin.inc.php#L190. 
It seems like the update from the plugin gallery did not work correctly, I am just wondering how that would be possible :-O 
The web user would definitely need to be able to write to the plugins/generic/ folder i.e. for the old folder to be removed and the new folder to be unpacked there by the plugin gallery script... but however the system should throw the message... I will take a look into the plugin gallery code to see how it is functioning... but maybe someone else would have another idea... ?

**Comment by bozana** [comment]
And, as @NateWr said, the fix is only available for 3.3.0-4 and 3.3.0-5. For other versions the plugin gallery should not offer the upgrade. 
Also, the plugin is/was not available for the version 3.2.x at all. Thus, I assume those using 3.2.x would still have the old and for 3.2.x not functioning plugin version :-P -- maybe the plugin gallery should/could behave differently and at least inform the users for such cases?
Thus, I assume those using 3.2.x should update to 3.3.0-5. The fix/changes in the plugin version 1.0.2 needed some changes in the pkp-lib that are only available since 3.3.0.4.

**Comment by asmecher** [comment]
I think this is resolved with the 3.3.0-x release of the plugin.
